docs/kb/types.md

```diff
@@ -0,0 +1,46 @@
+---
+order: 14
+description: Starport scaffolding types
+---
+
+# Starport scaffolding types
+
+Starport can support different types, like a number, string, bool, etc. This document shows all current types supported into the `starport`.
+
+## Built-in Types
+
+| Type         | Alias    | Code Type   | Description                      |
+| ------------ | -------- | ----------- | -------------------------------- |
+| string       | -        | string      | Text type                        |
+| array.string | strings  | []string    | List of text type                |
+| bool         | -        | bool        | Boolean type                     |
+| int          | -        | int32       | Integer numbers                  |
+| array.int    | ints     | []int32     | List of integer numbers          |
+| uint         | -        | uint64      | Unsigned integer numbers         |
+| array.uint   | uints    | []uint64    | List of unsigned integer numbers |
+| coin         | -        | sdk.Coin    | Cosmos-sdk coin type             |
+| array.coin   | coins    | sdk.Coins   | List of Cosmos-sdk coin type     |
+
+## Custom Type Scaffold
+
+Starport gives the to use custom field scaffolded before into the chain. The developer can create a `list` type called `user` and use this type in the next scaffold type.
+
+### Scaffolding
+
+We will scaffold a new `Coordinator` type to be reusable in the future:
+```shell
+starport scaffold list coordinator address:string id:uint --no-message
+```
+Now we can scaffold a message using the `Coordinator` type:
+```shell
+starport scaffold message blocks owner:Coordinator approve:bool
```

Comment:
I don't know if it's the best example. In this case, it can be ambiguous because you would preferably just pass the coordinator ID to the message.

I think the best usage of custom type is nested types. I would just show a command
```
starport s type CoordinatorDescription ...
starport s list Coordinator desc:CoordinatorDescription
```
WDYT?